### DISCUSSION TITLE
Long timeseries support: thinking beyond pandas datetime range

### DISCUSSION THREAD

**Discussion by dhirendrajnu** [georgebv/pyextremes#60] [General]
Hi 

I have been trying to use the 700 years worth of time series and the pandas apparently supports at max 584 years for datetime with the unit ns. I was trying to compute the return value stability plot (using pyextremesv2.3.0, and a POT approach) and getting the following error: 

![image](https://github.com/georgebv/pyextremes/assets/29814555/94f045c4-5bf0-497b-9739-cf627976a9b3)

Is there any workaround for this? I understand that the function takes a pandas series with the date-time object as its index. 

my series looks like this:
![image](https://github.com/georgebv/pyextremes/assets/29814555/0b7a33e2-8934-4557-b1f8-dfd3d2a7200b)

Any help is appreciated. 

Kind regards, 
Dhirendra

**Comment by georgebv** [comment]
Wow, this is certainly something I've never seen before with pandas. One thing you could do (which is a hack) is to compress your time series - represent records as offsets relative to first timestamps and then reduce those offsets X times (e.g. 10). Then when you ask for a return period of '36.5D' you are actually looking at 1yr return period.

I don't know how to do it without a hack unfortunately - seems like a fundamental limitation of the way pandas represents timestamps.

**Comment by dhirendrajnu** [comment]
That's really helpful. But what happens in case of the estimated return period using the get_return_periods? do I need to multiply the estimated return periods with the used offset? because the estimated return periods using this approach are quite smaller. I expect them to be 3.6 times larger roughly but not sure if I am correct?

**Comment by georgebv** [comment]
You are correct. If you scale your time series, you should scale your return periods too. E.g., if you "squeeze" your time series 10x this means you need to "unsqueeze" your return periods 10x too - 1 year would be 36.5 days

**Comment by Filston37** [comment]
The date-time requirement is also a bit limiting if you are working with shorter time scales that are dated. I am working with data which is from a 3-hour simulation, and I'd like to use the POT to work out the 3h return period value (MPM and Expected). Is there a work around for this?

**Reply by georgebv** [reply]
Doesn't it work as is? Pandas should be fine down to nanoseconds. If this doesn't work, please post your example
